superchain/superchain_test.go

```diff
@@ -281,3 +281,23 @@ func TestContractBytecodes(t *testing.T) {
 		}
 	}
 }
+
+// TestCanyonTimestampOnBlockBoundary asserts that Canyon will activate on a block's timestamp.
+// This is critical because the create2Deployer only activates on a block's timestamp.
+func TestCanyonTimestampOnBlockBoundary(t *testing.T) {
+	for superchainName, superchainConfig := range Superchains {
+		if superchainConfig.Config.CanyonTime == nil {
+			continue
+		}
+		ct := *superchainConfig.Config.CanyonTime
+		for _, id := range superchainConfig.ChainIDs {
+			chainCfg := OPChains[id]
+			canyonOffset := ct - chainCfg.Genesis.L2Time
+			// Block time is hardcoded in op-node/rollup/superchain.go
+			if canyonOffset%2 != 0 {
```

Comment:
Not every chain genesis time may align perfectly with the hardfork time of every other chain in the same superchain target (although, if every chain has a standard genesis based on L1 time starting block time, it might, because the L1 time alignment).

Reply:
That's true and a problem for canyon. Hopefully it works for canyon b/c everyone chose a even start time. Maybe we want to be more explicit about this in the future.